src/color-gamut/color-gamut.js

```diff
@@ -10,162 +11,111 @@ const Self = class ColorGamut extends HTMLElement {
 		let styleURL = new URL(`./${ Self.tagName }.css`, import.meta.url);
 		this.shadowRoot.innerHTML = `
 			<style>@import url("${ styleURL }")</style>
-			<span id="label" part="label"></span>
+			<slot>
+				<span id="label" part="label"></span>
+			</slot>
 		`;
 
-		this.gamuts ??= this.getAttribute("gamuts") ?? this.constructor.defaultGamuts;
-	}
-
-	connectedCallback () {
-		this.#initialize();
-		this.#render();
-	}
-
-	#initialized;
-	#initialize () {
-		if (this.#initialized) {
-			return;
-		}
-
-		this.#initialized = true;
-
-		let textContent = this.textContent;
-
-		if (textContent) {
-			this.color ??= this.textContent;
-		}
-	}
-
-	#render () {
-		if (!this.isConnected) {
-			return;
-		}
-
 		if (!this.#label) {
 			this.#label = this.shadowRoot.querySelector("#label");
 		}
-
-		this.#updateGamut();
-	}
-
-	#gamut;
-	get gamut () {
-		return this.#gamut?.id;
 	}
 
 	get gamutLabel () {
-		return this.#gamut?.label ?? "";
+		return this.gamutInfo?.label ?? "";
 	}
 
-	#gamuts;
-	get gamuts () {
-		return this.#gamuts;
-	}
-
-	set gamuts (gamuts) {
-
-		this.#gamuts = this.constructor.parseGamuts(gamuts);
-	}
-
-	#updateGamut () {
-		if (!this.#color) {
-			this.#setGamut();
-			return;
-		}
-
-		let gamut = this.gamuts.find(gamut => gamut.id === "none" || this.#color.inGamut(gamut.id));
-		this.#setGamut(gamut);
-	}
-
-	#setGamut (gamut) {
-		let oldGamut = this.#gamut;
-
-		if (gamut === oldGamut) {
-			return;
+	propChangedCallback ({name, prop, detail: change}) {
+		if (name === "gamuts") {
+			this.style.setProperty("--gamut-count", this.gamuts.length - 1);
 		}
 
-		this.#gamut = gamut;
-		this.#label.textContent = this.#gamut?.label ?? "";
-
-		if (!gamut) {
-			this.removeAttribute("gamut");
-			this.removeAttribute("level");
-			this.style.removeProperty("--gamut-level");
-
-			return;
-		}
-
-		this.setAttribute("gamut", gamut.id);
-		this.setAttribute("level", gamut.level);
-		this.style.setProperty("--gamut-level", gamut.level);
-
-		this.dispatchEvent(new CustomEvent("gamutchange", {
-			detail: {oldGamut, gamut},
-			bubbles: true,
-		}));
-	}
-
-	#color;
-	get color () {
-		return this.#color;
-	}
-
-	set color (color) {
-		this.#color = Color.get(color);
-		this.#render();
-	}
-
-	static observedAttributes = ["gamuts"];
-
-	attributeChangedCallback (name, oldValue, newValue) {
-		if (!name && this.hasAttribute("gamuts") || name === "gamuts") {
-			newValue ??= this.getAttribute("gamuts");
-
-			if (oldValue !== newValue) {
-				this.gamuts = this.constructor.parseGamuts(newValue ?? this.constructor.defaultGamuts);
-				this.#render();
+		if (name === "gamut") {
+			if (this.gamutInfo) {
+				this.style.setProperty("--gamut-level", this.gamutInfo.level);
+				this.style.setProperty("--gamut-label", this.gamutInfo.label);
+				this.style.setProperty("--gamut-id", `"${ this.gamutInfo.id }"`);
```

Comment:
I would also suggest storing `this.gamutInfo` in a variable so that the corresponding getter is not invoked four times in a row.

Reply:
Oh, right. Forgot it. 😅